### DISCUSSION TITLE
How the "Lock touchpad" feature work?

### DISCUSSION THREAD

**Discussion by mickoz** [timschneeb/GalaxyBudsClient#484] [Q&A]
Today I decided to use my "**Galaxy Buds2 Pro**" with my computer (bad headaches/pressures/etc. and my headphones are putting too much pressure and my softer ones are half-broken right now, and I like the comfort of my Galaxy Buds2 Pro usually, even if I keep them exclusively for my phone on the go).

Usually I disable "touch control" on my phone (I might like the idea... however, I touch too much my earbud there and there that it become annoying to me if it start/stop the music as it is very sensitive).  However as you know, on Windows it will have default settings and so touch control were enabled.  And that is where I found "GalaxyBudsClient" to the rescue!

I have a problem to understand the "Lock touchpad" feature, and even wonder if it works as expected?

On the Galaxy Wear app, it is very intuitive.  It is named "Touch controls" and there is a toggle to put it on/off.  Off = no touch, on = touch controls enabled.

In GalaxyBudsClient, it is named "Lock touchpad" and maybe I have not found the right resource, but I fail to understand what "Lock" mean.

I have played with it.  Sometime when it is "on" it seems to disable the "Touch controls", and sometime when it is "off" it seems to disable the "Touch control" and vice-versa.  It seems that only toggling change the state when I come back to the screen.  But its status does not seem to reflect anything coherent.

Also I expect the Left action and Right action to be the action that is executed when I touch it if enabled.  However, it is set to volume and it actually apply Play/Stop when enabled... so doesn't work as I expect at least.

The Double-tap earbud edge option to turn the volume down/up seems to work fine when enabled!

Now, all these behaviors that seem incoherent to me.  Are they a bug, or is it me that totally not understand the feature and/or the way it is presented that is counter-intuitive (I think I read elsewhere there are other options that influence this behavior, but logically it should all be in that screen and easy to understand).

Can you explain to me the intention of all this or if you think it is a kind of bug?

Why not take the same approach as the Galaxy Wear app and put a control that let you enable/disable the "Touch controls" and showing the status? (maybe there is a technical reason and why it is named "Lock Touchpad")

P.S., If there is any help that explain all this let me know.  I searched... Maybe too much headaches today and I end up being dumb dumb? :P

Thanks

**Comment by timschneeb** [comment]
Hi, the current version has a few bugs with the Galaxy Buds2 Pro & Buds FE because I didn't have any hardware to test with.
I'm currently doing a major app rewrite and solving some compatibility issues with newer earbud models. I recently borrowed a pair of Buds2 Pro, so I can verify that it works correctly.

I tested it using the new version I'm working on, and I haven't noticed any inconsistencies with the touch controls anymore, so it should be fixed in the next version.

The toggle is named 'Lock touchpad' because the official manager app also used to do it this way. I didn't notice that they inverted their 'lock touch' toggle to an 'enable touch' toggle in their app until now.

The new app will grey-out all the touch-related settings when the touchpad is locked, so it should be much clearer what it does:
![image](https://github.com/ThePBone/GalaxyBudsClient/assets/38386967/b948e50a-d4cb-4e6e-8cda-bd4cc28a06a5)
